### PR TITLE
fix: Remove deleted codecov Python package dependency

### LINKED ISSUE #568: Drop Deprecated CodeCov library
# Problem
The codecov library has been deprecated and now yanked from pypi. Our builds are all broken

# Approach
Replace with the [CodeCov Uploader GitHub Action](https://github.com/codecov/codecov-action) and remove from all of the poetry projects

### PR DESCRIPTION
**Pull request by matthewfeickert**
Resolves #568 

* On 2023-04-12 Codecov deleted the codecov package from PyPI. As the package no longer exists any attempt to install the 'test' extra dependencies will fail.
   - c.f. https://community.codecov.com/t/codecov-yanked-from-pypi-all-versions/4259
* Add requests to transformer_sidecar test dependencies as it is required by the test suite and was previously provided by codecov.
* Rebuild poetry lock files with `poetry lock --no-update`.
* Use codecov/codecov-action GitHub Action to upload coverage report.

With PR applied:

```console
$ git grep "codecov"
.github/workflows/ci_servicex.yaml:        uses: codecov/codecov-action@v3
code_generator_funcadl_uproot/.github/workflows/ci.yaml:        uses: codecov/codecov-action@v3
code_generator_funcadl_uproot/README.md:[![Code Coverage](https://codecov.io/gh/ssl-hep/ServiceX_Code_Generator_FuncADL_uproot/graph/badge.svg)](https://codecov.io/gh/ssl-hep/ServiceX_Code_Generator_FuncADL_uproot)
code_generator_funcadl_xAOD/.github/workflows/ci.yaml:        uses: codecov/codecov-action@v3
code_generator_funcadl_xAOD/README.md:[![Code Coverage](https://codecov.io/gh/ssl-hep/ServiceX_Code_Generator_FuncADL_xAOD/graph/badge.svg)](https://codecov.io/gh/ssl-hep/ServiceX_Code_Generator_FuncADL_xAOD)
did_finder_rucio/.github/workflows/ci.yaml:      - name: Report coverage using codecov
did_finder_rucio/.github/workflows/ci.yaml:        uses: codecov/codecov-action@v3
did_finder_rucio/README.md:[![codecov](https://codecov.io/gh/ssl-hep/ServiceX_DID_Finder_Rucio/branch/master/graph/badge.svg?token=xLpoqlrdE3)](https://codecov.io/gh/ssl-hep/ServiceX_DID_Finder_Rucio)
servicex_app/.github/workflows/ci.yaml:        uses: codecov/codecov-action@v3
servicex_app/README.rst:.. image:: https://codecov.io/gh/ssl-hep/ServiceX_App/branch/develop/graph/badge.svg
servicex_app/README.rst:  :target: https://codecov.io/gh/ssl-hep/ServiceX_App
```

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@BenGalewsky can you approve the CI to run for this?

**Comment by matthewfeickert** [comment]
Ah, cool. I hadn't properly poked at the CI so I didn't realize it was just a lint check. :) 

This is ready for review now too. Happy to answer any questions.

**Comment by BenGalewsky** [comment]
> Ah, cool. I hadn't properly poked at the CI so I didn't realize it was just a lint check. :)

That's one job... the real job did fail
https://github.com/ssl-hep/ServiceX/actions/runs/4681204490

**Comment by matthewfeickert** [comment]
> The workflow is not valid. .github/workflows/ci_servicex.yaml (Line: 73, Col: 9): Unexpected value 'uses' .github/workflows/ci_servicex.yaml (Line: 74, Col: 9): Unexpected value 'with'

Ah, I forgot to remove the `working-directory`.

**Comment by matthewfeickert** [comment]
```pytb
_______________ ERROR collecting tests/test_servicex_adapter.py ________________
ImportError while importing test module '/home/runner/work/ServiceX/ServiceX/transformer_sidecar/tests/test_servicex_adapter.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_servicex_adapter.py:30: in <module>
    from transformer_sidecar.servicex_adapter import ServiceXAdapter
src/transformer_sidecar/servicex_adapter.py:30: in <module>
    import requests
E   ModuleNotFoundError: No module named 'requests'

```

Hm, `requests` got killed along with `codecov` for some reason (dependency probably). Let me add that back.

**Comment by matthewfeickert** [comment]
Okay that's passing in a local Docker container now

```console
(venv) root@b8c4b281c19d:/workdir/transformer_sidecar# poetry run python -m coverage run -m pytest -r sx
================================================================================= test session starts =================================================================================
platform linux -- Python 3.10.11, pytest-7.2.1, pluggy-1.0.0
rootdir: /workdir/transformer_sidecar, configfile: pytest.ini
plugins: mock-3.10.0, flask-1.2.0
collected 19 items                                                                                                                                                                    

tests/test_object_store_manager.py ...                                                                                                                                          [ 15%]
tests/test_object_store_uploader.py ..                                                                                                                                          [ 26%]
tests/test_rabbit_mq_manager.py .                                                                                                                                               [ 31%]
tests/test_servicex_adapter.py ...                                                                                                                                              [ 47%]
tests/test_transformer_argument_parser.py ...                                                                                                                                   [ 63%]
tests/test_watched_directory.py ...                                                                                                                                             [ 78%]
tests/transformer_stats/test_aod_stats.py ..                                                                                                                                    [ 89%]
tests/transformer_stats/test_uproot_stats.py ..                                                                                                                                 [100%]

================================================================================= 19 passed in 20.31s =================================================================================

```

so that should hopefully pass CI as it was the only one failing earlier.

File: transformer_sidecar/pyproject.toml [hunk 1]
```diff
@@ -23,10 +23,10 @@ optional = true
 [tool.poetry.group.test.dependencies]
 flake8 = "^5.0.4"
 pytest-mock = "^3.10.0"
-codecov = "^2.1.12"
 pytest = "^7.2.0"
 pytest-flask = "^1.2.0"
 coverage = "^6.5.0"
+requests = "^2.28.2"
```
Comment (matthewfeickert) [inline transformer_sidecar/pyproject.toml:29]: Adding here as `requests` was apparently a `codecov` dependency and is required for the `pytest` test suite to pass.

**Review by BenGalewsky — APPROVED** [review]
Thank you so much for your contribution!

**Pull request merged**